iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_ptr.inl

```diff
@@ -105,7 +107,7 @@ inline void unique_ptr<T>::reset(T* const ptr) noexcept
 template <typename T>
 inline void unique_ptr<T>::destroy() noexcept
 {
-    if (m_ptr && m_deleter)
+    if (m_ptr)
     {
         m_deleter(m_ptr);
     }
```

Comment:
The use after move warning would happen one level above and a dtor is not really a use after move ... or maybe I got you wrong here...

Reply:
No, it is just a surprise that clang-tidy does not wrongly assume this is a use after move since `m_deleter` is moved but also not accessed in the moved from object.